onnxruntime/core/providers/coreml/builders/impl/conv_op_builder.cc

```diff
@@ -46,18 +46,57 @@ Status ConvOpBuilder::AddToModelBuilderImpl(ModelBuilder& model_builder, const N
   std::unique_ptr<COREML_SPEC::NeuralNetworkLayer> layer = CreateNNLayer(model_builder, node);
 
   const auto& input_defs = node.InputDefs();
+  const auto& output_defs = node.OutputDefs();
+  const auto& input_name = input_defs[0]->Name();
+  const auto& output_name = output_defs[0]->Name();
 
   const auto& weight_tensor = *model_builder.GetInitializerTensors().at(input_defs[1]->Name());
-  const auto& weight_shape = weight_tensor.dims();
+  auto weight_tensor_prime = weight_tensor;
+  auto& weight_shape = weight_tensor_prime.dims();
+
+  const bool is_1d_conv = (weight_shape.size() == 3);
+
+  if (is_1d_conv) {
+    // weight_tensor needs to be expanded from MXCXH->MXCXHx1
+    weight_tensor_prime.add_dims(1);
+  }
 
   NodeAttrHelper helper(node);
-  const auto strides = helper.Get("strides", std::vector<int64_t>{1, 1});
-  const auto onnx_pads = helper.Get("pads", std::vector<int64_t>{0, 0, 0, 0});
-  const auto dilations = helper.Get("dilations", std::vector<int64_t>{1, 1});
+  auto strides = helper.Get("strides", std::vector<int64_t>{1, 1});
+  auto dilations = helper.Get("dilations", std::vector<int64_t>{1, 1});
+  auto onnx_pads = helper.Get("pads", std::vector<int64_t>{0, 0, 0, 0});
+  // Strides/dilations for 1d conv is normally of length 1. Expand them by 1
+  // to meet the required length 2 (for 2d conv it's normally 2)
+  // Similarly 1d conv normally has a length 2 padding. Expand it to length 4 by adding additional zeros.
+  if (is_1d_conv) {
+    if (strides.size() < 2) {
+      strides.push_back(1);
+    }
+    if (dilations.size() < 2) {
+      dilations.push_back(1);
+    }
+    if (onnx_pads.size() < 4) {
+      onnx_pads.insert(onnx_pads.begin() + 1, 0);
```

Comment:
>onnx_pads.insert(onnx_pads.begin() + 1, 0); [](http://example.com/codeflow?start=6&length=43)

the comment above says we append zeros, if I understand correctly.
this inserts 0 at index 1, right?
how about appending zeros with `onnx_pads.insert(onnx_pads.end(), 4 - onnx_pads.size(), 0)`?